.circleci/scripts/find_changed_packages.js

```diff
@@ -0,0 +1,120 @@
+/*
+ *                      Copyright 2024 Salto Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { execSync, exec } = require('child_process')
+const { readFileSync, writeFileSync, existsSync, readdirSync } = require('fs')
+const path = require('path')
+
+const getChangedFiles = (userInputBaseCommit) => {
+  const commitHashRegex = /\b[0-9a-f]{9}/g
+  const baseCommit = userInputBaseCommit ?? execSync('git merge-base main HEAD').toString().trim()
+  console.log('base commit:', baseCommit)
+  const output = execSync(`git diff --oneline --name-only ${baseCommit}..HEAD`).toString().split('\n')
+  console.log('git diff output:', output)
+  return output.filter(file => file && file.startsWith('packages/'))
+}
+
+const getWorkspacesInfo = () => {
+  const output = execSync('yarn --json workspaces info', { encoding: 'utf8' })
+  return JSON.parse(JSON.parse(output).data)
+}
+
+const resolveTransitiveDependencies = (packageName, workspaceInfo) => {
+  const packageInfo = workspaceInfo[packageName]
+  if (!packageInfo) {
+    return
+  }
+
+  return packageInfo['workspaceDependencies'].map(dep => resolveTransitiveDependencies(dep, workspaceInfo))
+}
+
+
+const generateDependencyMapping = (workspaceInfo) => {
+  return Object.entries(workspaceInfo)
+  .map(([packageName, info]) => 
+    info.workspaceDependencies.map(dep => [dep, packageName])
+  )
+  .flat()
+  .reduce((acc, [dep, packageName]) => {
+    if (!acc[dep]) {
+      acc[dep] = []
+    }
+    acc[dep].push(packageName)
+    return acc
+  }, {})
+}
+
+const hasE2eTests = (packageName) => {
+  const e2eDir = path.join(__dirname, '..', '..', 'packages', packageName, 'e2e_test')
+  const e2eDirExists = existsSync(e2eDir)
+  const e2eDirHasTestFiles = e2eDirExists && readdirSync(e2eDir).some(file => file.endsWith('.test.ts'))
+  console.log('e2eDir:', e2eDir, 'exists:', e2eDirExists, 'hasTestFiles:', e2eDirHasTestFiles)
+  return e2eDirHasTestFiles
+}
+
+const findChangedPackages = (userInputBaseCommit, workspaceInfo) => {
+  const changedFiles = getChangedFiles(userInputBaseCommit)
+  console.log('Changed files:', changedFiles)
+  const changedPackageLocation = changedFiles.map(path => path.split('/').slice(0, 2).join('/')).filter((value, index, self) => self.indexOf(value) === index)
+
+  const changedPackages = changedPackageLocation.map(packageDir =>Object.keys(workspaceInfo).find(key => workspaceInfo[key].location === packageDir))
```

Comment:
any chance this file is not covered by any linter?
```suggestion
  const changedPackages = changedPackageLocation.map(packageDir => Object.keys(workspaceInfo).find(key => workspaceInfo[key].location === packageDir))
```